wlauto/workloads/googleslides/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,469 @@
+        timer = new Timer();
+        timer.start();
+        clickView(BY_DESC, "New presentation");
+        clickView(BY_TEXT, "New PowerPoint", true);
+        timer.end();
+        results.put("create_document", timer);
+        endDumpsys(ACTIVITY_DOCLIST, "create_document");
+
+        // first slide
+        enterTextInSlide("Title", "WORKLOAD AUTOMATION");
+        enterTextInSlide("Subtitle", "Measuring perfomance of different productivity apps on Android OS");
+        saveDocument(docName);
+
+        insertSlide("Title and Content");
+        enterTextInSlide("title", "Extensions - Workloads");
+        enterTextInSlide("Text placeholder", SLIDE_TEXT_CONTENT);
+        clickView(BY_DESC, "Text placeholder");
+        clickView(BY_DESC, "Format");
+        clickView(BY_TEXT, "Droid Sans");
+        clickView(BY_TEXT, "Droid Sans Mono");
+        clickView(BY_ID, PACKAGE_ID + "palette_back_button");
+        UiObject decreaseFont = getViewByDesc("Decrease text");
+        repeatClickView(decreaseFont, 20);
+        getUiDevice().pressBack();
+
+        // get image from gallery and insert
+        insertSlide("Title Only");
+        clickView(BY_DESC, "Insert");
+        clickView(BY_TEXT, "Image", true);
+        clickView(BY_TEXT, "Recent");
+        try {
+            UiObject image = new UiObject(new UiSelector().resourceId("com.android.documentsui:id/date").instance(2));
+            image.clickAndWaitForNewWindow();
+        } catch (UiObjectNotFoundException e) {
+            clickView(BY_ID, "com.android.documentsui:id/date", true);
+        }
+
+        // last slide
+        insertSlide("Title Slide");
+        // insert "?" shape
+        clickView(BY_DESC, "Insert");
+        clickView(BY_TEXT, "Shape");
+        clickView(BY_TEXT, "Buttons");
+        clickView(BY_DESC, "actionButtonHelp");
+        UiObject resize = getViewByDesc("Bottom-left resize");
+        UiObject shape = getViewByDesc("actionButtonHelp");
+        UiObject subtitle = getViewByDesc("subTitle");
+        resize.dragTo(subtitle, 40);
+        shape.dragTo(subtitle, 40);
+        enterTextInSlide("title", "THE END. QUESTIONS?");
+
+        sleep(1);
+        getUiDevice().pressBack();
+        deleteDocument(docName);
+    }
+
+    public void insertSlide(String slideLayout) throws Exception {
+        sleep(1); // a bit of time to see previous slide
+        UiObject view = getViewByDesc("Insert slide");
+        view.clickAndWaitForNewWindow();
+        view = getViewByText(slideLayout);
+        view.clickAndWaitForNewWindow();
+    }
+
+    public void enterTextInSlide(String viewName, String textToEnter) throws Exception {
+        UiObject view = getViewByDesc(viewName);
+        view.click();
+        view.setText(textToEnter);
+        try {
+            clickView(BY_DESC, "Done");
+        } catch (UiObjectNotFoundException e) {
+            clickView(BY_ID, "android:id/action_mode_close_button");
+        }
+        SystemClock.sleep(200);
```

Comment:
Introduces a small wait time for stability (not measured). On some devices, keyboard pops up when entering text, and takes a finite amount of time to disappear after clicking Done. In these cases, trying to find a view immediately after entering text leads to `UiObjectNotFoundException`